pkg/fieldpath/fieldpath.go

```diff
@@ -26,18 +27,23 @@ import (
 )
 
 // FormatMap formats map[string]string to a string.
-func FormatMap(m map[string]string) (fmtStr string) {
+func FormatMap(m map[string]string) string {
```

Comment:
would adding a benchmark test here help show the perf optimization?

Reply:
> would adding a benchmark test here help show the perf optimization?



```
goos: darwin
goarch: amd64
pkg: k8s.io/kubernetes/pkg/fieldpath
cpu: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
BenchmarkFormatMap-12            3010065               395.4 ns/op           128 B/op          7 allocs/op
BenchmarkFormatMapOld-12         1595634               771.6 ns/op           176 B/op          9 allocs/op
PASS
ok      k8s.io/kubernetes/pkg/fieldpath 4.027s

```

Benchmark code
```go
func BenchmarkFormatMap(b *testing.B) {
	m := map[string]string{"foo": "bar", "abc": "ddd"}
	b.ReportAllocs()
	b.ResetTimer()
	for n := 0; n < b.N; n++ {
		FormatMap(m)
	}
}

func BenchmarkFormatMapOld(b *testing.B) {
	m := map[string]string{"foo": "bar", "abc": "ddd"}
	b.ReportAllocs()
	b.ResetTimer()
	for n := 0; n < b.N; n++ {
		FormatMapOld(m)
	}
}
```

``` 
// FormatMap formats map[string]string to a string.
func FormatMapOld(m map[string]string) (fmtStr string) {
	// output with keys in sorted order to provide stable output
	keys := sets.NewString()
	for key := range m {
		keys.Insert(key)
	}
	for _, key := range keys.List() {
		fmtStr += fmt.Sprintf("%v=%q\n", key, m[key])
	}
	fmtStr = strings.TrimSuffix(fmtStr, "\n")

	return
}
```

Don't know if such a benchmark is OK
@jsturtevant

Reply:
>Don't know if such a benchmark is OK

In what sense?

It helps to show that there was an improvement by making the change.  Can even add the new test into the test suite: https://github.com/kubernetes/community/blob/4026287dc3a2d16762353b62ca2fe4b80682960a/contributors/devel/sig-testing/testing.md#benchmark-unit-tests

Reply:
> > Don't know if such a benchmark is OK
> 
> In what sense?
> 
> It helps to show that there was an improvement by making the change. Can even add the new test into the test suite: https://github.com/kubernetes/community/blob/4026287dc3a2d16762353b62ca2fe4b80682960a/contributors/devel/sig-testing/testing.md#benchmark-unit-tests

I added the benchmark to the test file，Please have a look again